php/libraries/Candidate.class.inc

```diff
@@ -200,7 +201,7 @@ class Candidate
      */
     static public function createNew(
         int $centerID,
-        string $dateOfBirth,
+        ?string $dateOfBirth,
```

Comment:
The error checking is good but I don't think the type of this param should change. I think DoB is always required? So this should be reverted and line 229 can be deleted also.

Reply:
@johnsaigle DoB isn't always required.

Reply:
Why did you open this PR when the changes are already in the other one?

Reply:
@johnsaigle I thought it would be best to pull the code out and into a separate PR because people complain about PRs from me not being condensed. Fate doesn't seem to want me to win in any case. 🥺

Reply:
I think it's case-by-case. That other PR looks small and manageable but it's not passing Travis.

Reply:
@johnsaigle I'm not good at judging the case-by-case. I think this PR is good because it fixes a problem that should be fixed. The other PR changes something else but I ran into this problem when working on it and so I had it in it. So I thought today it would be good to pull it into a separate PR. I never have luck with modifying the test code and so I don't think that test code will be fixed anytime soon.

Reply:
> I'm not good at judging the case-by-case.

Fair enough. Maybe an explanation would be helpful?

The idea behind preferring PRs to be small and targeted to one issue is that it makes it easier to review. An ideal PR changes only a small number of things and does so clearly, with easy-to-follow testing instructions and documentation.

If you're being asked to make a PR smaller it's usually because someone is having a hard time reviewing it OR that it's doing many things that will make it more difficult to track changes going forward. 

> So I thought today it would be good to pull it into a separate PR. I never have luck with modifying the test code and so I don't think that test code will be fixed anytime soon

This seems like a different discussion. I don't think anyone is ignoring that PR because it's too complicated. It's one of the smaller PRs that are open. However, it's flagged as Needs Work and is not passing Travis. To me, that signals that it doesn't need my attention because the ball is in your court to finish the work and test it using our automated tools.

I think a better investment of time would be to debug the test code. If you're not comfortable doing so or get stuck, you should ask for help from someone (or use the "Help! I don't understand Travis!" flag). Otherwise, the PR can't progress.

Reply:
```
This seems like a different discussion. I don't think anyone is ignoring that PR because it's too complicated. It's one of the smaller PRs that are open. However, it's flagged as Needs Work and is not passing Travis. To me, that signals that it doesn't need my attention because the ball is in your court to finish the work and test it using our automated tools.

I think a better investment of time would be to debug the test code. If you're not comfortable doing so or get stuck, you should ask for help from someone (or use the "Help! I don't understand Travis!" flag). Otherwise, the PR can't progress.
```

@johnsaigle can a new tag be made for differentiating `Needs work` from a `tests are just failing` tag. I don't like writing or modifying the test code at all because it never works when it should from my last attempts at it for PRs where I've tried. I've also been told to wait for new test code changes for react. So a lot of my PRs when tests don't go right just kept left aside.

On a side note I think this PR should be separate from the other PRs. More so than previous PRs where people have commented.

Reply:
What I'm trying to say is that it's helpful if you communicate your intentions to the team so that we can all work together on problems. 

If something is marked as Needs Work and isn't passing the testing suite then there's no reason for anyone to think that that pull request is in a state to be reviewed. 

It'll be easier to keep things from getting left aside if you use something like the "Help" tag or "Blocked" in case the PR needs new React testing changes.